### DISCUSSION TITLE
NUnit Framework

### DISCUSSION THREAD

**Discussion by Aryan-sharma11** [Aryan-sharma11/AddSevenPS#10] [General]
Used the NUnit framework for Unit testing 

Referred material  https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-fsharp-with-nunit

The test only focuses on the F sharp functions  right now

**Comment by Aryan-sharma11** [comment]
> Used the NUnit framework for Unit testing
> 
> Referred material https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-fsharp-with-nunit
> 
> The test only focusses on the F sharp functions only right now

Attributes I used are :

- [ TestFixture ] - denotes a class that contains a test

- [ Test ] -  marks the beginning of a test

Assert - used to test an expression, upon a failure in Debug mode, an assertion generates a system error dialog box

**Comment by Aryan-sharma11** [comment]
NUnit vs XUnit

 - xUnit is much more extensible when compared to NUnit 

 - The [Fact] attribute is used instead of the [Test] attribute.
 
 -  [Fact] attribute to implement non parameterized tests, whereas the [Theory] attribute is used for parameterized tests
 
https://www.lambdatest.com/blog/nunit-vs-xunit-vs-mstest/#:~:text=As%20far%20as%20NUnit%20vs,of%20the%20%5BTest%5D%20attribute.

**Comment by abhi18av** [comment]
Okay, this sounds good.

> xUnit is much more extensible when compared to NUnit


Then shall we continue with `xunit` for the time being?

**Comment by Aryan-sharma11** [comment]
Okay I added the XUnit framework too


> Okay, this sounds good.
> 
> > xUnit is much more extensible when compared to NUnit
> 
> Then shall we continue with `xunit` for the time being?
